crates/services/upgradable-executor/src/executor.rs

```diff
@@ -798,7 +822,43 @@ where
                         .to_string(),
                 ))
             }
-        }
+        }?;
+
+        let execution_result = result.result();
+        let iter = execution_result.tx_status.iter();
+        let iter = iter.zip(execution_result.block.transactions());
+        let mut preconfirmations = iter
+            .enumerate()
+            .map(|(i, (status, tx))| {
+                let tx_index = u16::try_from(i).unwrap_or(u16::MAX);
+                let preconfirmation_status =
+                    convert_tx_execution_result_to_preconfirmation(
+                        tx,
+                        status.id,
+                        &status.result,
+                        *execution_result.block.header().height(),
+                        tx_index,
+                    );
+                preconfirmation_status
+            })
+            .collect::<Vec<_>>();
+
+        let squeezed_out =
+            execution_result
+                .skipped_transactions
+                .iter()
+                .map(|(tx_id, error)| Preconfirmation {
+                    tx_id: *tx_id,
+                    status: PreconfirmationStatus::SqueezedOut {
+                        reason: error.to_string(),
+                    },
+                });
+
+        preconfirmations.extend(squeezed_out);
+
+        let _ = preconfirmation_sender.send(preconfirmations).await;
```

Comment:
Does this really get us much? Like this happens after the block is executed, so maybe just a little bit. I feel like maybe it's okay if wasm executors don't support pre-confs.

Reply:
This PR removes `notify_skipped_transaction` from PoA, so only the way how TxPool will clean up transaction are preconfirmations right now.

Plus later, we can add support preconfirmations for the WASM as well. In this case it will not hurt the network during the upgrade